examples/api/test/widgets/heroes/hero.2_test.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/heroes/hero.2.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Hero flight animation with additional overlay', (WidgetTester tester) async {
```

Comment:
This is basically a copy of `hero.0_test.dart`. I created it because I suppose I'm required to create a test for every sample. This test doesn't check for behavior of the additional overlay because I can't think of a way to check "paint order" using the `WidgetTester`. Alternative would be to test it in a white-box manner, like I did in `heroes_test.dart` but that a) would be kinda pointless b) would require unnecessary modifications to the sample. What do you think?

Reply:
If the test doesn't test the purpose of the example, the test is pretty pointless. 

One thing you could do is perform a hit test at a location where the two overlap and ensure they are hit in the expected order.